localstack/services/cloudwatch/cloudwatch_database_helper.py

```diff
@@ -0,0 +1,407 @@
+                    if metric.get("Timestamp")
+                    else _get_current_unix_timestamp_utc()
+                )
+
+                inserts = []
+                if metric.get("Value") is not None:
+                    inserts.append({"Value": metric.get("Value"), "TimesToInsert": 1})
+                elif metric.get("Values"):
+                    counts = metric.get("Counts", [1] * len(metric.get("Values")))
+                    inserts = [
+                        {"Value": value, "TimesToInsert": int(counts[indexValue])}
+                        for indexValue, value in enumerate(metric.get("Values"))
+                    ]
+
+                for insert in inserts:
+                    for _ in range(insert.get("TimesToInsert")):
+                        cur.execute(
+                            self._get_insert_single_metric_query(),
+                            (
+                                account_id,
+                                region,
+                                metric.get("MetricName"),
+                                namespace,
+                                unix_timestamp,
+                                self._get_ordered_dimensions_with_separator(
+                                    metric.get("Dimensions")
+                                ),
+                                metric.get("Unit"),
+                                metric.get("StorageResolution"),
+                                insert.get("Value"),
+                            ),
+                        )
+
+                if statistic_values := metric.get("StatisticValues"):
+                    cur.execute(
+                        self._get_insert_aggregated_metric_query(),
+                        (
+                            account_id,
+                            region,
+                            metric.get("MetricName"),
+                            namespace,
+                            unix_timestamp,
+                            self._get_ordered_dimensions_with_separator(metric.get("Dimensions")),
+                            metric.get("Unit"),
+                            metric.get("StorageResolution"),
+                            statistic_values.get("SampleCount"),
+                            statistic_values.get("Sum"),
+                            statistic_values.get("Minimum"),
+                            statistic_values.get("Maximum"),
+                        ),
+                    )
+
+            conn.commit()
+
+    def get_metric_data_stat(
+        self,
+        account_id: str,
+        region: str,
+        query: MetricDataQuery,
+        start_time: datetime,
+        end_time: datetime,
+        scan_by: str,
+    ):
+        # TODO exclude null values, check if dimensions must be null though if missing
+        # TODO test default order
+        # TODO add filters for dimensions
+        # TODO add filters for unit
```

Comment:
please add these TODOs also as remaining tasks for the PR 🙏 we can tackle those issues in upcoming PRs. 
but being able to handle optional Namespace, and dimensions would be important 🙂